app/src/forms/email/reminderService.js

```diff
@@ -0,0 +1,296 @@
+      return EmailTypes.REMINDER_FORM_NOT_FILL;
+    }
+
+    const yend_date = moment(end_date).subtract(1, 'day');
+
+    if(moment(now).isSame(yend_date)){
+      return EmailTypes.REMINDER_FORM_WILL_CLOSE;
+    }
+
+    return state;
+  },
+  checkIfInMiddleOfThePeriod : ( now, start_date,  days_diff )=>{
+    if (days_diff < 6 ) return false;
+    let interval = Math.floor(days_diff/2);
+    // eslint-disable-next-line no-console
+    let mail_date = moment(start_date).add(interval, 'days').format('YYYY-MM-DD');
+    return moment(now).isSame(mail_date);
+  },
+  _runQueries : async (obj) => {
+
+    obj.fillers = [];
+    await UserFormAccess.query()
+      .select('formVersionId','formName','userId','firstName','lastName', 'email')
+      .where('formId', obj.form.id)
+      .modify('filterActive', true)
+      .modify('filterByAccess', undefined, Roles.FORM_SUBMITTER, undefined)
+      .modify('orderDefault').then(function(data) {
+        obj.submitters = data;
+      });
+
+    if(obj.submitters && obj.submitters.length>0){
+      await SubmissionData.query()
+        .select('confirmationId', 'createdAt', 'submissionId', 'formVersionId','userId','firstName','lastName', 'email')
+        .where('formId',obj.form.id)
+        .modify('filterDrafts', false)
+        .modify('filterDeleted', false)
+        .modify('filterCreatedAt', obj.report.dates.startDate, obj.report.dates.graceDate)
+        .modify('orderDefault').then(function(data2) {
+          obj.fillers = (data2) ? data2 : [];
+        });
+    }
+
+    return  obj;
+  },
+  getDifference : async (array1, array2) => {
+    return array1.filter(object1 => {
+      return !array2.some(object2 => {
+        return object1.userId === object2.userId;
+      });
+    });
+  },
+  _initStatement : async (obj) => {
+    const statement = {};
+    statement.form = obj.form;
+    statement.report = obj.report;
+    statement.state = obj.state;
+    statement.submitters = [];
+
+    if(!obj.submitters || obj.submitters.length == 0) return statement;
+
+    if(statement.state != EmailTypes.REMINDER_FORM_OPEN ){
+      if(obj.fillers &&  obj.fillers.length != 0 ) {
+        statement.submitters = await service.getDifference(obj.submitters, obj.fillers);
+      } else {
+        statement.submitters = obj.submitters;
+      }
+
+    } else {
+      statement.submitters = obj.submitters;
+    }
+    return statement;
+  },
+  _getReferer : () => {
+    // We create this function because in the header we cant get the real referer but
+    // this function allow us to generate the referer dynamicly
+    try {
+      const protocol = 'https://';
+      const basePath = config.get('frontend.basePath');
+      const host = process.env.SERVER_HOST;
+      return`${protocol}${host}${basePath}`;
+    } catch (error) {
+      log.error(error.message, {
+        function: '_getReferer'
+      });
+      throw error;
+    }
+  },
+  _initMaillSender: async (statement, referer) => {
```

Comment:
This should be `_initMailSender: async(statement, referer) => {`

Reply:
correct now

Reply:
I solved it